.github/actions/setup-dev-container/action.yml

```
@@ -78,4 +82,14 @@ runs:
             /home/${DEVCONTAINER_USER}/.local \
             /home/${DEVCONTAINER_USER}/.gradle \
           && . ./dev-env.sh \
-          && workspace-install-affected"
+          && workspace-install-nodejs-dependencies"
+
+    - name: Get affected Nx projects
+      id: get-affected-projects
+      shell: bash
+      run: |
+        AFFECTED_PROJECTS=$(devcontainer exec --workspace-folder ../sage-monorepo bash -c "
+          . ./dev-env.sh && nx show projects --affected --sep ','")
+        echo "Affected projects: $AFFECTED_PROJECTS"
+        echo "affected_projects=$AFFECTED_PROJECTS" >> $GITHUB_ENV
+        echo "::set-output name=affected_projects::$AFFECTED_PROJECTS"
```

Per the [GitHub blog](https://github.blog/changelog/2023-07-24-github-actions-update-on-save-state-and-set-output-commands/) -- 

> Workflows using save-state or set-output in their workflows will continue to work as expected, however, a warning will appear under annotations indicating the planned deprecation. We recommend customers using these commands to upgrade their workflows to use environment files.

We could consider switching to `GITHUB_OUTPUT` as described [here](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/): 
```suggestion
        echo "affected_projects=$AFFECTED_PROJECTS" >> $GITHUB_OUTPUT
```